h2/src/main/org/h2/util/DateTimeUtils.java

```diff
@@ -1470,5 +1470,49 @@ public static long convertScale(long nanosOfDay, int scale) {
         }
         return nanosOfDay - mod;
     }
+    
+    /**
+     * Truncate the given date to 'day'
+     * 
+     * @param timeUnit the time unit (e.g. 'DAY', 'HOUR', etc.)
+     * @param value the date
+     * @return date truncated to 'day'
+     */
+    public static Value truncateDate(String timeUnit, Value value) {
+        Value result = null;
```

Comment:
I think that it's better not to assign null value to variable here, but only declare variable.

```Java
Value result;
```

In this case compiler will generate an error if some of branches below do not set an value or throw an exception.

Reply:
I will update this in a new PR. Thanks !